.github/workflows/advance-zq.yml

```diff
@@ -1,12 +1,143 @@
 name: Advance ZQ
 # This type must match the event type from zq.
 # https://help.github.com/en/actions/reference/events-that-trigger-workflows#external-events-repository_dispatch
+# These events only trigger on the Github default branch (usually
+# master).
 on:
   repository_dispatch:
     types: [zq-pr-merged]
 jobs:
   advance-zq:
     name: Advance ZQ
-    runs-on: ubuntu-18.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node-version: [12.x]
+        os: [ubuntu-18.04]
     steps:
+      # Only one of these should run at a time, and the checkout of brim
+      # has to be in the "protected section".
+      - name: Turnstyle
+        uses: softprops/turnstyle@v0.1.2
```

Comment:
I couldn't tell from the github site for this action, is there a default timeout value?

Reply:
Sort of. The step allows "timeout and continue". Timeout and fail is configurable via https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes . While I gave this action some basic testing, it's true it's not rigorously been tested in our environment. To avoid the chance of a big queue building up due to some bug, I've set `timeout-minutes` to 30 from its default of 360.